### PR TITLE
Add block schema in proper format

### PR DESCRIPTION
**Pull request by matthewjablack**

### CODE REVIEW

File: schema/block.json [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      "type": "string",
+      "title": "Previous Block Hash",
+      "pattern": "^0x([A-Fa-f0-9]{64})$"
+    },
+    "nonce": {
+      "type": "number",
+      "title": "Nonce",
+      "minimum": 0
+    },
+    "_raw": {
```
Comment (monokh) [inline schema/block.json]: Thought we agreed defining the properties in `_raw` wasn't really necessary? @mattBlackDesign @harshjv

File: schema/block.json [hunk 2]
```diff
@@ -0,0 +1,201 @@
+{
+  "$id": "https://dev.liquality.com/schema/block.json",
+  "title": "Block",
+  "description": "Blockchain block",
+  "type": "object",
+  "required": [ "number", "hash", "timestamp", "difficulty", "size", "previousBlockHash", "nonce" ],
```
Comment (monokh) [inline schema/block.json]: what would `nonce` be in bitcoin?

Comment (monokh) [reply schema/block.json]: are these limited to chain commonalities? notice `size` in there too

File: schema/block.json [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        "gasLimit": {
+          "type": "number",
+          "title": "Gas Limit",
+          "minimum": 0
+        },
+        "gasUsed": {
+          "type": "number",
+          "title": "Gas Used",
+          "minimum": 0
+        },
```
Comment (harshjv) [inline schema/block.json]: SyntaxError! 🖌

File: schema/block.json (hunk 1 above)
Comment (harshjv) [reply schema/block.json]: @mattBlackDesign according to yesterday's discussion, we are just focusing on commonalities. You can safely remove `properties` of `_raw`. just make it `object`.

Comment (matthewjablack) [reply schema/block.json]: Updated

File: schema/block.json (hunk 2 above)
Comment (matthewjablack) [reply schema/block.json]: `nonce` in btc is used for POW in the process of mining

Yes, these are chain commonalities

File: schema/block.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      "type": "string",
+      "title": "Parent Hash",
+      "pattern": "^0x([A-Fa-f0-9]{64})$"
+    },
+    "previousBlockHash": {
+      "type": "string",
+      "title": "Previous Block Hash",
+      "pattern": "^0x([A-Fa-f0-9]{64})$"
+    },
+    "nextBlockHash": {
```
Comment (harshjv) [inline schema/block.json]: If `parentHash` is here, then what is the use of `previousBlockHash`?

File: schema/block.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      "type": "number",
+      "title": "Total Difficulty",
+      "minimum": 0
+    },
+    "extraData": {
+      "type": "string",
+      "title": "Extra Data",
+      "pattern": "^0x([A-Fa-f0-9]{64})$"
+    },
+    "gasLimit": {
```
Comment (monokh) [inline schema/block.json]: the schema should only have properties defined for chain commonalities 
When the data is constructed, everything else can go in the `_raw` property - which doesn't need defining what properties it contains. its just a passthrough of all the chain specific properties that the chain may have.

Comment (matthewjablack) [reply schema/block.json]: Agreed, added in following format:
```
"_raw": {
  "type": "object"
}
```

File: schema/block.json (hunk 4 above)
Comment (matthewjablack) [reply schema/block.json]: Updated to use `parentHash`

**Comment by matthewjablack** [comment]
Updated @harshjv. Would love to get re-review before merging.

**Comment by harshjv** [comment]
@mattBlackDesign can you move this file to `src/schema`? (PS. do rebase this before you push :) )

**Comment by harshjv** [comment]
@mattBlackDesign did you rebase it?

**Comment by monokh** [comment]
Github shows no changes? Is this branch and master really in sync? 😄

**Pull request closed without merging**